pkg/controller/gittrack/gittrack_controller.go

```diff
@@ -103,6 +104,11 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	err = c.Watch(&source.Kind{Type: &farosv1alpha2.ClusterGitTrack{}}, &handler.EnqueueRequestForObject{})
```

Comment:
So far, the different cases have only been checked in the function that ignores the resources if their namespace doesn't match, the controllers would still watch all resources.

According to my comment about the namespaced mode, we would have to watch all resources.

Reply:
I'd like to be able to switch off the watching of everything as in certain modes it shouldn't be necessary and, with fewer watches, would allow people to reduce their RBAC scope too.

> According to my comment about the namespaced mode, we would have to watch all resources.

My thought here was that you will likely have multiple "namespaced" instances of the controller (one per namespace) and then a separate "cluster" instance running to manage the cluster scoped resources. So the individual namespaced controllers should be able to completely ignore `ClusterGitTracks` and `ClusterGitTrackObjects` in theory unless they are running both the `Namespaced` and `ClusterScoped` modes within the same controller instance